src/commands/generate/__tests__/cliTest.ts

```diff
@@ -99,13 +99,58 @@ describe("generate command", () => {
             );
         });
 
+        it("generate packageJson with productDependencies with specified optional", async () => {
+            const productDependencies: ISlsManifestDependency[] = [
```

Comment:
nit: Async tests typically have:

```js
expect.hasAssertions();
```

To catch scenarios when an `expect` invocation accidentally getting lost in the callback of a free function that never fires. By default Jest tests pass if `expect` is never called and the callback doesn't throw an exception.

https://jestjs.io/docs/asynchronous#asyncawait